### ISSUE TITLE
Using multiple unloaders fixed order of calling?

### ISSUE THREAD

**Issue by Yasp0** [Stephan-S/FS22_AutoDrive#140]
When using multiple unloaders is (or will) it possible to set an order of calling for the unloader?

For example, I use 3 unloaders on a large sugarbeet field. It would be nice if first unloader 1 will be 100% loaded, then 2 and then 3. Instead of just calling the closest(?) unloader. Now sometimes  for example all 3 unloaders are 75% full.

**Comment by Stephan-S** [comment]
Yeah I would like to look into it and make sure that it stores the last called unloader, so that the harvester will stick with that one if it is still available.

**Comment by rohne83** [comment]
Looks to me like #713 and #663 are exactly about the same issue.

**Comment by rohne83** [comment]
Since the other two issues are closed already, seems this one will now be the main issue for that topic.
In general, calling order of unloaders by distance can make sense on very large fields, but in very most situations, by filllevel would be the more appropriate approach.
Due to the current priority based solely on distance, we regularly face issues especially when chopping: for unknown reason, the assignment of current unloader to the chopper seems to get lost for short time during many chopper turns, and once the chopper arrives at the start of the next row to continue its course, the unloader gets called again. If in that situation the unloader waiting at the field edge is a few meters closer to the chopper than the original unloader (which might even have a filllevel of 70-90%, only missing a few liters until he can go to the silo) that one will be called instead of completing filling of the original unloader. This will cause headaches on the field with several unloaders standing around, and also eventually result in very long waiting times for the chopper as most unloaders are filled about the same time and none left for continue chopping. Both problems wouldn't occur at all if priority was by fill-level instead of by distance.
So I'd see 3 different solution approaches: 1. make the priority setting configurable (by distance, by fill-level, default should be the latter), or if you don't want to have an additional setting, then 2. make it solely dependent on fill-level (the situations where by distance would be more appropriate are very rare) or 3. use the algorithm that CP is using, which is mainly based on filllevel but also takes into account the distance to a quite good degree it seems.
